db/db_ut.c

```diff
@@ -416,46 +416,35 @@ int db_print_set(const db_con_t* _c, char* _b, const int _l, const db_key_t* _k,
 int db_is_neq_type(db_type_t _t0, db_type_t _t1)
 {
 	// LM_DBG("t0=%d t1=%d!\n", _t0, _t1);
-	if(_t0 == _t1)
-		return 0;
-
-	switch(_t1) {
-	case DB_INT:
-		if(_t0==DB_BIGINT || _t0==DB_DATETIME || _t0==DB_BITMAP)
-			return 0;
-		break;
-
-	case DB_STRING:
-		if(_t0==DB_STR || _t0 == DB_BLOB)
-			return 0;
-		break;
-
-	case DB_STR:
-		if(_t0==DB_STRING || _t0==DB_BLOB)
-			return 0;
-		break;
-
+	switch(_t0) {
 	case DB_BIGINT:
-		if(_t0==DB_INT || _t0==DB_DATETIME || _t0==DB_BITMAP)
-			return 0;
-		break;
-
-	case DB_BLOB:
-		if(_t0==DB_STR || _t0==DB_STRING)
-			return 0;
-		break;
-
+	case DB_BITMAP:
 	case DB_DATETIME:
-		if(_t0==DB_INT || _t0==DB_BIGINT || _t0==DB_BITMAP)
+	case DB_INT:
+		switch(_t1) {
+		case DB_BIGINT:
+		case DB_BITMAP:
+		case DB_DATETIME:
+		case DB_INT:
 			return 0;
-		break;
-
-	case DB_BITMAP:
-		if(_t0==DB_INT || _t0==DB_BIGINT || _t0==DB_DATETIME)
+		default:
+			break;
+		}
+	case DB_BLOB:
+	case DB_STR:
+	case DB_STRING:
+		switch(_t1) {
+		case DB_BLOB:
+		case DB_STR:
+		case DB_STRING:
 			return 0;
-		break;
-
+		default:
+			break;
+		}
 	default:
+		if(_t0 == _t1) {
+			return 0;
+		}
 		break;
 	}
 
```

Comment:
Personal preference here, not sure if you agree.  Wouldn't this code block read better if it were moved below into a single line, with the end of the function simply saying: `return _t0 != _t1;`?

Reply:
Agreed.